BrushedMotorControllerFirmware/include/driver.c

```diff
@@ -220,25 +274,52 @@ static void continuous_calculation_timer_handler(struct k_timer *dummy)
 K_TIMER_DEFINE(continuous_calculation_timer, continuous_calculation_timer_handler, NULL);
 #pragma endregion TimerWorkCallback
 
-// encoder functions
-static void enc_callback(enum ChannelNumber chnl, enum PinNumber pin)
+bool is_target_achieved(enum ChannelNumber chnl)
 {
-	// TODO - maybe split this callback to two callbacks, for P1 and P0:
-	// what if motor is stuck and hitting one enc. from two directions?
-	// How will speed calculation behave if motor will keep changing position?
-	// (ex. when setting position?)
-	drv_chnls[chnl].count_cycles += 1;
+	uint32_t delta_shortest_path;
+
+	CALC_POSITION_DELTA(chnl)
 
-	if (drv_chnls[chnl].actual_mrpm == 0) {
-		drv_chnls[chnl].actual_dir = STOPPED;
-		return;
+	if (drv_chnls[chnl].position_delta < HALF_SPIN_DEGREES) {
+		delta_shortest_path = drv_chnls[chnl].position_delta;
+	} else {
+		delta_shortest_path = ((int32_t)FULL_SPIN_DEGREES -
+					drv_chnls[chnl].position_delta);
 	}
 
+	if (drv_chnls[chnl].target_achieved) {
+		if (delta_shortest_path <=
+			(FULL_SPIN_DEGREES) / (CONFIG_POS_CONTROL_PRECISION_MODIFIER)) {
+			return true;
+		}
+	} else {
+		if (delta_shortest_path <=
+		   ((FULL_SPIN_DEGREES * CONFIG_POS_CONTROL_HISTERESIS_PERCENTAGE) /
+		    (CONFIG_POS_CONTROL_PRECISION_MODIFIER * 100))) {
+
+			return true;
+		}
+	}
```

Comment:
```suggestion
	if ((drv_chnls[chnl].target_achieved &&
	     delta_shortest_path <= (FULL_SPIN_DEGREES) /
	     (CONFIG_POS_CONTROL_PRECISION_MODIFIER)) ||
	     (!drv_chnls[chnl].target_achieved &&
		       delta_shortest_path <= ((FULL_SPIN_DEGREES * CONFIG_POS_CONTROL_HISTERESIS_PERCENTAGE) /
		    (CONFIG_POS_CONTROL_PRECISION_MODIFIER * 100))))) {
			return true;
		}
```

Reply:
Resolved slightly different in https://github.com/eSqadron/NRF52840-Brushed-Motor-Controller-Firmware/pull/13, https://github.com/eSqadron/NRF52840-Brushed-Motor-Controller-Firmware/pull/13/commits/efb6ec49a8be358916778c0422bfc54958570225, @maciekbaczmanski please check if resolution is acceptable